app/src/main/java/org/simple/clinic/enterotp/EnterOtpUpdate.kt

```diff
@@ -22,6 +24,14 @@ class EnterOtpUpdate(
       UserVerifiedInBackground -> dispatch(GoBack)
       is RequestLoginOtpCompleted -> requestOtpCompleted(model, event)
       is EnterOtpResendSmsClicked -> next(model.requestLoginOtpStarted(), RequestLoginOtp as EnterOtpEffect)
+      is OtpEntryProtectedStateChanged -> effectsForStateChanged(event.stateChanged)
+    }
+  }
+
+  private fun effectsForStateChanged(stateChanged: ProtectedState): Next<EnterOtpModel, EnterOtpEffect> {
+    return when (stateChanged) {
+      is Allowed -> dispatch(AllowOtpEntry)
+      else -> // do something
```

Comment:
Correct me if I am wrong, depending on the protected state we dispatch either allow or block effects right? Instead of having 2 effects for changing the OTP entry mode. Why not pass the `OtpEntryMode` to a single effect from here? 🤔 

Like this
```kt
val otpEntryMode = when (stateChanged) {
  // do something
}
return dispatch(ChangeOtpEntryMode(otpEntryMode))
```

While this is working with UI effects, I think ideally this should be done in UI renderer.

Reply:
Ah makes sense, updating this to be included in UiRenderer